WinBackend has drifted 7 methods behind the Backend trait (plus a stale draw_data_table signature) — cargo check --features win fails on develop

## Summary

`cargo check --features win` does not compile on `develop`. `WinBackend`
(`quadraui/src/win/backend.rs`) has drifted behind the `Backend` trait by seven
methods and carries one stale signature.

Measured on `develop` @ `5f06867`, 2026-08-07:

```
error[E0050]: method `draw_data_table` has 3 parameters but the declaration in
              trait `Backend::draw_data_table` has 4
error[E0046]: not all trait items implemented, missing:
              `draw_settings_chrome`, `status_bar_layout`, `tab_bar_layout`,
              `activity_bar_layout`, `draw_command_line`, `draw_chart`,
              `chart_layout`
error: could not compile `quadraui` (lib) due to 2 previous errors
```

## Why it went unnoticed

Nothing compiles this feature. Not the coord Test stage
(`cargo build --features tui --features gtk --features terminal`), not `cargo test
--features tui`, and not CI — whose jobs are `gtk`, `tui`, and `rustfmt` only. So
`win/backend.rs` can drift arbitrarily far without any signal.

The most recent addition is from **today**: #531 introduced `draw_settings_chrome`
as a required (no-default) trait method and merged without a `WinBackend`
implementation. That is not a criticism of #531 — there was no mechanism that could
have told anyone.

## Scope

Bring `WinBackend` back to a compiling state against the current trait:

- implement the seven missing methods as `todo!()` stubs, consistent with the file's
  existing convention (its module doc: *"Every method is a `todo!()` stub. Implement
  against Direct2D / DirectWrite (via `windows-rs`) and the compiler tells you when
  you're done."*);
- correct `draw_data_table` to the trait's 4-parameter signature.

This is deliberately **stub-level parity, not a real Direct2D implementation.** The
goal is to restore the compiler's ability to report drift. Actual rasterisers are
milestone #40 / `docs/CROSS_PLATFORM.md` work.

`win = []` has zero dependencies and the backend is pure stubs, so this compiles and
is verifiable **on Linux** — no Windows runner or hardware required.

## Acceptance

- [ ] `cargo check --features win` exits 0 on Linux.
- [ ] Every new method is a `todo!()` stub matching the file's existing style; no
      method silently returns a plausible-but-wrong value.
- [ ] `draw_data_table`'s signature matches the trait exactly.
- [ ] No change to any other backend, and no default impl added to the trait
      (`PRIMITIVE_RULES.md` rule 7, `docs/SMELL_AUDIT_2026-07.md` PORT-01).

## Note

`draw_terminal_divider` is **not** part of this backlog — #533 implements it in
`WinBackend` correctly. Verified against that branch: it is absent from the E0046
missing list. If #533 lands first, do not re-add it.

Blocks the `win` half of the compile-gate issue: that gate cannot go green until this
lands.

--- a/quadraui/src/win/backend.rs
+++ b/quadraui/src/win/backend.rs
@@ -171,6 +171,7 @@ impl Backend for WinBackend {
         &mut self,
         _rect: Rect,
         _table: &crate::DataTable,
+        _hovered_idx: Option<usize>,
     ) -> crate::DataTableLayout {
         todo!("Direct2D data table rasteriser")
     }
@@ -193,6 +194,17 @@ impl Backend for WinBackend {
 
     fn draw_palette(&mut self, _rect: Rect, _palette: &Palette) {
         todo!("Direct2D palette rasteriser")
+    }
+
+    fn draw_settings_chrome(
+        &mut self,
+        _rect: Rect,
+        _header_text: &str,
+        _query: &str,
+        _placeholder: &str,
+        _active: bool,
+    ) {
+        todo!("Direct2D settings chrome rasteriser")
     }
 
     fn draw_status_bar(
@@ -223,12 +235,32 @@ impl Backend for WinBackend {
         todo!("Direct2D activity bar rasteriser")
     }
 
+    fn status_bar_layout(&self, _rect: Rect, _bar: &StatusBar) -> StatusBarLayout {
+        todo!("DirectWrite status bar layout")
+    }
+
+    fn tab_bar_layout(&self, _rect: Rect, _bar: &TabBar) -> TabBarHits {
+        todo!("DirectWrite tab bar layout")
+    }
+
+    fn activity_bar_layout(&self, _rect: Rect, _bar: &ActivityBar) -> Vec<ActivityBarRowHit> {
+        todo!("DirectWrite activity bar layout")
+    }
+
     fn draw_terminal(&mut self, _rect: Rect, _term: &Terminal) {
         todo!("Direct2D terminal cell grid rasteriser")
     }
 
     fn draw_text_display(&mut self, _rect: Rect, _td: &TextDisplay) {
         todo!("Direct2D text display rasteriser")
+    }
+
+    fn draw_command_line(
+        &mut self,
+        _rect: Rect,
+        _cmd: &crate::primitives::command_line::CommandLine,
+    ) {
+        todo!("Direct2D command line rasteriser")
     }
 
     fn text_display_layout(&self, _rect: Rect, _td: &TextDisplay) -> TextDisplayLayout {
@@ -445,5 +477,23 @@ impl Backend for WinBackend {
         _view: &crate::primitives::diff_view::DiffView,
     ) -> crate::primitives::diff_view::DiffViewLayout {
         todo!("Direct2D DiffView rasteriser")
+    }
+
+    fn draw_chart(
+        &mut self,
+        _rect: Rect,
+        _chart: &crate::primitives::chart::Chart,
+        _hovered_point: Option<(usize, usize)>,
+        _crosshair_x: Option<f64>,
+    ) -> crate::primitives::chart::ChartLayout {
+        todo!("Direct2D chart rasteriser")
+    }
+
+    fn chart_layout(
+        &self,
+        _rect: Rect,
+        _chart: &crate::primitives::chart::Chart,
+    ) -> crate::primitives::chart::ChartLayout {
+        todo!("DirectWrite chart layout")
     }
 }

